src/app/shared/components/course-form/course-form.component.ts

```diff
@@ -31,8 +34,11 @@ export class CourseFormComponent implements OnInit {
   courseForm!: FormGroup;
   fullAuthorArray!: AuthorModel[];
   courseAuthorArray!: AuthorModel[];
-  submitted = false;
-  newAuthorSubmitted = false;
+  submitted: boolean = false;
+  newAuthorSubmitted: boolean = false;
+
+  courseId: string = "";
+  course!: CourseModel | null;
 
   ngOnInit(): void {
```

Comment:
The OnInit method is expanding from line 43 to 86. This is way too long and should be separated into smaller, individual chunks. For example:
`ngOnOnint(): void { this.buildForm(); this.initAuthors(); }`